docs/debugging.md

```diff
@@ -0,0 +1,117 @@
+
+The required dependencies are:
+
+- Docker:
+  - Mac: [Docker Desktop](https://www.docker.com/products/docker-desktop)
+  - Ubuntu/Debian: `sudo apt install docker docker-compose`
+- Python 3 (recommended 3.6 or 3.7)
+  - It is recommended to use a virtual environment to install Substra packages (for instance [virtualenv](https://virtualenv.pypa.io/en/latest/))
+- Substra package
+
+If you want to debug your script using assets from the Substra platform, you also need access to the platform the assets are on.
+
+
+## Run your code locally
+
+With this installation, you can write scripts using the [SDK](../references/sdk.md) as you would for a remote execution, then
+set the `debug` parameter to `True` when creating the client.
+
+```python
+client = substra.Client.from_config_file(profile_name="node-1", debug=True)
+```
+
+### Use assets from a deployed Substra platform and local assets
+
+With this, you gain access to the platform in 'read-only' mode and any asset you create is created locally.
+
+This means that any function to get, describe or download an asset works with assets from the platform and local assets. Functions to
+list assets will list the assets from the platform and the local ones.  
+Functions that create a new asset (they start with `add`) only create local assets.  
+To differentiate between a local asset and a remote one, look at their key (or pkhash): for local assets, it starts with `local_`.
+
+Local assets are saved in-memory, so they are deleted at the end of the script.
+
+### Dependency between assets
+
+Any asset created locally can depend on an asset from the platform or on another local asset:
+- a traintuple can use an algo and dataset from the platform
+- a testtuple can use an objective from the platform
+- etc.
+
+There is one exception: a traintuple, testtuple or any other tuple created locally cannot depend on a tuple from the platform 
+or from a compute plan defined on the platform. This is impossible because it is forbidden to download models from the platform.
+
+Since no data can leave the platform, a task depending on a dataset from the platform uses the fake data generated from the opener. The
+number of fake samples generated is equal to the number of samples the task would have used.
+
+### Debugging with logs and breakpoints
+
+#### Access the logs
+
+In debug mode, the logs of the execution of the traintuples and testtuples are saved
+in the `log` field.
+
+To display them:
+
+```python
+traintuple = client.add_traintuple(...)
+print(traintuple["log"])
+```
+
+Moreover, if there is an error in the execution, those logs are displayed in the call stack.  
+So any `print` or warning in the algo file for example will be displayed.
+
+#### Debug with pdb
+
+The execution is done synchronously, so the script waits for the task in progress to end before continuing.  
+The execution of the tuples happens in Docker containers that are spawned on the fly and removed once the execution is done.
```

Comment:
I think these 2 lines are really important and should be in a more common section (in "Run your code locally" maybe?).

Reply:
OK !